ui/app/src/components/FolderBrowserTreeView/FolderBrowserTreeView.tsx

```diff
@@ -14,135 +14,239 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React, { useEffect, useMemo, useRef, useState } from 'react';
-import { getIndividualPaths } from '../../utils/path';
-import FolderBrowserTreeViewUI, { FolderBrowserTreeViewNode } from './FolderBrowserTreeViewUI';
+import useSpreadState from '../../hooks/useSpreadState';
 import LookupTable from '../../models/LookupTable';
-import { ApiResponse } from '../../models/ApiResponse';
-import { forkJoin, Observable } from 'rxjs';
-import { useActiveSiteId } from '../../hooks/useActiveSiteId';
-import { useLogicResource } from '../../hooks/useLogicResource';
-import Suspencified from '../Suspencified/Suspencified';
-import FolderBrowserTreeViewSkeleton from './FolderBrowserTreeViewSkeleton';
-import { LegacyItem } from '../../models/Item';
-import { fetchLegacyItemsTree } from '../../services/content';
-import { legacyItemsToTreeNodes } from './utils';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { ApiResponse } from '../../models';
+import PathNavigatorSkeletonTree from '../PathNavigatorTree/PathNavigatorTreeSkeleton';
+import { fetchChildrenByPath, fetchItemByPath } from '../../services/content';
+import useActiveSite from '../../hooks/useActiveSite';
+import { PathNavigatorTreeItem, PathNavigatorTreeNode } from '../PathNavigatorTree';
+import { lookupItemByPath, parseSandBoxItemToDetailedItem } from '../../utils/content';
+import { ApiResponseErrorState } from '../ApiResponseErrorState';
+import TreeView from '@mui/lab/TreeView';
+import { createLookupTable, nnou } from '../../utils/object';
+import { isFolder } from '../PathNavigator/utils';
 
 export interface FolderBrowserTreeViewProps {
   rootPath: string;
-  initialPath?: string;
-  showPathTextBox?: boolean;
```

Comment:
What was this for? Why remove?

Reply:
Those props are not being used.
- initialPath: I don't think this is needed since file browse dialog always has the rootPath as the initial path.
- showPathTextBox: I don't think this is something the component should handle, since the containers are the ones showing the path (e.g. BrowseFilesDialog showing the current path at the top).